### PR TITLE
Add onboarding documentation

### LINKED ISSUE #92: Onboarding new developers requires too much explaination
Our README explains the problem space well, but it doesn't do much for onboarding. We should have more "onboarding" information that explains how to set up the project on a new machine, how to contribute, the technologies/tools we use, etc.

### PR DESCRIPTION
**Pull request by leesharma**
Resolves #92 

Major changes are as follows:
- `bin/setup` now includes admin initialization, so it can be used to set up new downloads
- `README.md` has two new sections:
  - Quick Start is designed to get a new user set up with a working environment and a copy of the repository
  - Contributing points to the guidelines and resources for project contributors
- `CONTRIBUTING.md` lays out our contribution guidelines, from the code of conduct to testing

This PR is parallel to some additions to the project Wiki for details that don't need to be included in the project source code.

### CODE REVIEW

**Comment by leesharma** [comment]
I can't sleep, so here's a pull request.

What do you all think about the README information? I'm planning on putting other details (git flow, development tools like guard, etc.) in the wiki. Anything else that needs to be on the front page?

What about the contribution guidelines? Too loose or tight? Are they confusing at all?

**Comment by jdsayle** [comment]
Hey all,

(Sorry I couldn't make it out yesterday!) @Lee, I think that the
contribution guidelines/readme are clearly written and easy to follow. Only
thing I'd say - and I know that it's a WIP at the moment - the wiki is
fairly barren. Other than that . . . WOW nice work!

Just my 2 cents,
-JD

**Comment by leesharma** [comment]
@jdsayle Thanks! I'll be fleshing out the wiki more today–yesterday was just a skeleton.

Also, to everyone: I have some questions about our git workflow.

So far, we've been doing a branch-based workflow with work being done on `<name>/<short desc of feature/bug>` (ex. `robb/actually-send-mail`). However, this breaks down when someone new wants to join the project–they can't claim issues or make new branches.

How do we want to handle newcomers to the project? Keep the same "internal" workflow and have newcomers use forks to start? When should they get repository access, and who has the permission to grant it?

**Comment by bjmllr** [comment]
These are great questions and they might be questions we want R4G to have an opinion on, just so we don't have to wrestle with them again every year. I'll share my thoughts.

I like forks. I think the single fork works better for groups where everyone is committed to the project and/or there is a certain degree of rapport (the "core team"), and multiple forks works better for newcomers and occasional contributors. Some of us have that rapport and/or commitment, but most OSS contributions are just spikes, so I think it makes sense to use both.

If someone outside the core team wants to claim an issue, I think it's enough to put a comment saying "I'll take this on" in the issue. This is probably worth calling out in `CONTRIBUTING.md`.

Re joining the core team: Most OSS contributions are just spikes, so I think it makes sense to invite someone in after they've shown that a pattern of commitment to the project's success. Also, a special responsibility of the core team is to review and merge contributions, so I think it makes sense to prefer people that have demonstrated they'll do this effectively. Also also ... hopefully this doesn't come up, but not everyone will meet our standards of niceness/good citizenship/conduct/etc, and we're representing R4G as members, so let's try only to invite people who will.

Generally the core team decides who joins the core team (subjectively, to account for the subjective nature of the criteria in the previous paragraph). I added @leesharma unilaterally, maybe I should have asked someone else first, though I don't think anyone would have pushed back. On the other hand, for a project and team this size, I suspect the stakes are low enough and the trust is high enough for the team leader to make those decisions.

I agree with all the file changes in this PR, and the wiki info looks good too.

**Comment by leesharma** [comment]
I added the line that @bjmllr described (about claiming issues) and filled in the wiki. Is there anything else to add/edit for in-repo documentation, or is this ready for merge?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+At this point you're waiting on us–we'll try to respond to your PR quickly.
+We may suggest some changes or improvements or alternatives.
+
+Some things that will increase the chance that your pull request is accepted.
+
+* Use Rails idioms and helpers
+* Include tests that fail without your code, and pass with it
+* Update the documentation, the surrounding one, examples elsewhere, guides,
+  whatever is affected by your contribution
+* The tests and rubocop should both pass–this is required for a successful build!
```
Comment (atzorvas) [inline CONTRIBUTING.md]: So bundle exec rake could report for rubocop results too? (maybe it's a silly idea, but the lesser the steps, the better I think it is)

**Comment by leesharma** [comment]
@atzorvas: that makes sense–I'll consolidate the rake tasks.

**Comment by leesharma** [comment]
Anything else to add/change, or is this ready for now?

**Comment by jdsayle** [comment]
Hi All,

I've been out of the loop for a bit (ie - sorry if this was addressed but
didn't see it in comments anywhere . . . ) I get a 404 error when clicking
on the Contribution Guide links in the wiki. If this is a known issue,
perhaps just putting (WIP) in text next to the links and on the
Contribution Guide page will work for now? At least until until we have
that page populated with content.

Other than that small thing I think that the wiki is clearly written and
covers everything I'd want to know or see.

-JD

**Comment by leesharma** [comment]
Oh, forgot to mention that! It links to the location `CONTRIBUTING.md` will be when it's merged into master. We could put "WIP" for now, but if the changes are ready for merge anyway, I'd push for that.

---
Lee Sharma
lee@leesharma.com 
(518) 635-0533

**Comment by leesharma** [comment]
Actually, I just went through and changed the links to point to the `lee/onboarding-docs` branch CONTRIBUTING.md for now (so there should be no more broken links); I can change them back when this does get merged.

**Comment by bjmllr** [comment]
I'd say this is ready to merge.

**Comment by leesharma** [comment]
Awesome!

Could someone else hit the "merge" button? I don't want to self-merge, especially since this PR includes warnings against merging your own PR.

**Comment by robbkidd** [comment]
👍 

I'll bungee in here, push the button, and bungee away again.

**Comment by leesharma** [comment]
:heart:

(Sadly, there's no bungee emoticon on GitHub.)

**Pull request merged**